common/app/common/NewNavigation.scala

```diff
@@ -498,19 +498,19 @@ object NewNavigation {
 
     def getSectionOrTagId(page: Page) = {
       val tags = Navigation.getTagsFromPage(page)
-      val commonKeywords = tagPages.intersect(tags.keywordIds)
+      val commonKeywords = tags.keywordIds.intersect(tagPages)
```

Comment:
Does the `intersect` modify the order of the initial `List`? I don't think so but I might be wrong.

If you still want to make it explicit you can do it like this, it is shorter 😁  `tags.keywordIds.intersect(tagPages).sortBy(tags.keywordIds.indexOf)`